pkg/component/component.go

```diff
@@ -20,6 +20,12 @@ import (
 
 const NotAvailable = "Not available"
 
+// Below are various "outputs" that we use in both the CLI and test cases, so we must
+// put them in a common place
+const InitializingNewComponent = "Initializing new component"
+const SourceCodeDetected = "Files: Source code detected, a Devfile will be determined based upon source code autodetection"
+const NoSourceCodeDetected = "Files: No source code detected, a starter project will be created in the current directory"
```

Comment:
It's to reduce copy-and-pasting in the tests :) So that's why I put them as constants.

Reply:
I agree with defining constants, but my point was more about moving those constants to `pkg/odo/cli/init/init.go` instead of  `pkg/component/component.go`.
I feel like those constants should belong to the `init` CLI layer.
But maybe I missed something about the codebase structure :D

Reply:
Ah, but if you try to import `pkg/odo/cli/init` you'll run into issues in the testing suite! So into `component.go` it goes.

Reply:
In this case, let's create a new package on `pkg/odo/cli`, for example `messages`, intended to declare all the messages strings that the tests could use. This way, you should not have cycled imports.

Reply:
> Ah, but if you try to import `pkg/odo/cli/init` you'll run into issues in the testing suite! So into `component.go` it goes.

I tried doing this yesterday, and Go was complaining that `init` needed to be a func. So I tried using an import alias instead, and the tests passed (no import cycle reported at that time).

> In this case, let's create a new package on `pkg/odo/cli`, for example `messages`, intended to declare all the messages strings that the tests could use. This way, you should not have cycled imports.

That would be a better alternative, I think.